client/src/controllers/DropdownController.ts

```diff
@@ -187,22 +194,78 @@ export class DropdownController extends Controller<HTMLElement> {
         if (hoverTooltipInstance) {
           hoverTooltipInstance.enable();
         }
+        onHide();
+      },
+    };
+  }
+
+  get useCustomHideOnClickAway() {
+    // Tippy's default hideOnClick option for "hiding on click away" doesn't
+    // work well with our datetime libraries. Instead, we use a custom plugin
+    // to hide the tooltip when clicking outside the dropdown. It's unlikely
+    // we'll render a datetime picker for themes other than drilldown, so use
+    // the custom solution for this theme only.
+    return this.themeValue === 'drilldown';
+  }
+
+  /**
+   * Hides tooltip on click away from the reference element.
+   * A custom implementation to replace tippy's default hideOnClick option,
+   * which doesn't play well with our datetime libraries (or others that render
+   * elements outside of the dropdown's DOM).
+   */
+  get hideTooltipOnClickAway() {
+    // We can't use `this` to access the controller instance inside the plugin
+    // function, so we need to get these ahead in time.
+    const reference = this.reference;
+    const toggleTarget = this.toggleTarget;
+
+    return {
+      name: 'hideTooltipOnClickAway',
+      fn(instance: Instance) {
+        const onClick = (e: MouseEvent) => {
+          if (
+            instance.state.isShown &&
+            // Hide if the click is outside of the reference element,
+            // or if the click is on the toggle button itself.
+            (!reference.contains(e.target as Node) ||
+              toggleTarget.contains(e.target as Node))
+          ) {
+            instance.hide();
+          }
+        };
+
+        return {
+          onShow() {
+            document.addEventListener('click', onClick);
+          },
+          onHide() {
+            document.removeEventListener('click', onClick);
+          },
+        };
       },
     };
   }
 
   get plugins() {
-    return [
+    const plugins = [
       hideTooltipOnBreadcrumbsChange,
       hideTooltipOnEsc,
       rotateToggleIcon,
-    ].concat(this.hideOnClickValue ? [hideTooltipOnClickInside] : []);
+    ];
+    if (this.hideOnClickValue) {
+      plugins.push(hideTooltipOnClickInside);
+    }
+    if (this.useCustomHideOnClickAway) {
+      plugins.push(this.hideTooltipOnClickAway);
+    }
```

Comment:
I’m puzzled we’d use getters for something like this. It makes it unclear how the plugins are defined, and this isn’t an API intended to be particularly elegant anyway.